src/main/java/org/openjdk/jextract/impl/TreeMaker.java

```diff
@@ -111,20 +124,23 @@ public Declaration createTree(Cursor c) {
     private Declaration createTreeInternal(Cursor c) {
         Position pos = CursorPosition.of(c);
         // dedup multiple declarations that point to the same source location
-        Declaration decl = declarationCache.get(pos);
-        if (decl == null) {
-            decl = switch (c.kind()) {
-                case EnumDecl -> createEnum(c);
-                case EnumConstantDecl -> createEnumConstant(c);
-                case FieldDecl -> createVar(c, Declaration.Variable.Kind.FIELD);
-                case ParmDecl -> createVar(c, Declaration.Variable.Kind.PARAMETER);
-                case FunctionDecl -> createFunction(c);
-                case StructDecl -> createRecord(c, Declaration.Scoped.Kind.STRUCT);
-                case UnionDecl -> createRecord(c, Declaration.Scoped.Kind.UNION);
-                case TypedefDecl -> createTypedef(c);
-                case VarDecl -> createVar(c, Declaration.Variable.Kind.GLOBAL);
-                default -> null; // skip
-            };
+        Optional<Declaration> cachedDecl = lookup(pos);
+        if (cachedDecl.isPresent()) {
+            return cachedDecl.get();
+        }
+        Declaration decl = switch (c.kind()) {
+            case EnumDecl -> createEnum(c);
+            case EnumConstantDecl -> createEnumConstant(c);
+            case FieldDecl -> createVar(c, Declaration.Variable.Kind.FIELD);
+            case ParmDecl -> createVar(c, Declaration.Variable.Kind.PARAMETER);
+            case FunctionDecl -> createFunction(c);
+            case StructDecl -> createRecord(c, Declaration.Scoped.Kind.STRUCT);
+            case UnionDecl -> createRecord(c, Declaration.Scoped.Kind.UNION);
+            case TypedefDecl -> createTypedef(c);
+            case VarDecl -> createVar(c, Declaration.Variable.Kind.GLOBAL);
+            default -> null; // skip
+        };
+        if (decl != null && pos != Position.NO_POSITION) {
```

Comment:
If position is NO_POSITION, we do not cache. But I wonder, if there's no position (which indicates some clang intrinsics like va_list), should we really even generate a declaration?

Reply:
There is a difference in the two approaches suggested above. Consider this example:

```
#include <stdarg.h>

struct Foo {
    int x;
    va_list v;
};
```

If we simply skip all declarations with NO_POSITION, we get a weird layout for Foo:

```
private static final MemoryLayout $LAYOUT = MemoryLayout.structLayout(
        JAVA_INT.withName("x"),
        MemoryLayout.paddingLayout(4),
        MemoryLayout.paddingLayout(24)
    ).withName("Foo");
```

Whereas with this patch we get this:

```
private static final MemoryLayout $LAYOUT = MemoryLayout.structLayout(
        JAVA_INT.withName("x"),
        MemoryLayout.paddingLayout(4),
        MemoryLayout.sequenceLayout(1,         MemoryLayout.structLayout(
                JAVA_INT.withName("gp_offset"),
                JAVA_INT.withName("fp_offset"),
                foo_h.C_POINTER.withName("overflow_arg_area"),
                foo_h.C_POINTER.withName("reg_save_area")
            ).withName("__va_list_tag")    ).withName("v")
    ).withName("Foo");
```

Both layouts are "correct". The struct interface won't have any accessor for the `v` field no matter which approach is taken (which makes sense, as we have no real declaration for `va_list`). While the second layout is more precise, I'm starting to think that the first approach is more honest: if we don't really have a cursor position for a declaration, and we cannot deduplicate, we cannot guarantee that declaration attributes won't be "lost" (as they might be added to different copies of the same underlying intrinsic cursor). WDYT?

Reply:
I think it may ultimately be preferable to keep as much of the layout intact, even if we don't generate accessors. That way, the user could do something with the layout themselves if they wanted to. I think maybe deduplicating based on `Position` only is not enough in the presence of intrinsics which do not have a position. But, we can have another look at better intrinsic support later as well. I think it's fine to just drop them for now.